gdal/ogr/ogrsf_frmts/pgeo/ogrpgeotablelayer.cpp

```diff
@@ -195,6 +195,63 @@ CPLErr OGRPGeoTableLayer::Initialize( const char *pszTableName,
 
     poFeatureDefn->SetGeomType( eOGRType );
 
+    // try to associate domains with fields
+    CPLODBCStatement oTableList( poSession );
```

Comment:
as we have already done that in OGRPGeoDataSource::Open(), couldn't we just have a OGRPGeoDataSource::m_bHasGdbItemsTable member and test it here ?

Reply:
That was my original approach, but since the layers are created by the datasource BEFORE we iterate over the tables list we haven't yet performed that table iteration at the time this code is called...

Reply:
looking a bit at the Open() code, couldn't we slighly modify it to:
- list all table name and store them in a set + set m_bHasGdbItemsTable
- instantiate spatial tables (unchanged)
- instantiate non-spatial tables (modified to iterate over the collected set of names)

Reply:
Sure, I can do that

Reply:
Ok, done